Graphics/Vulkan/VulkanGraphicsTexture.cpp

```diff
@@ -1,24 +1,34 @@
 // Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
 // for more information.
 
+#include <NovelRT/Exceptions/InitialisationFailureException.hpp>
+#include <NovelRT/Exceptions/InvalidOperationException.hpp>
 #include <NovelRT/Graphics/Vulkan/Utilities/TextureAddressMode.hpp>
 #include <NovelRT/Graphics/Vulkan/VulkanGraphicsDevice.hpp>
 #include <NovelRT/Graphics/Vulkan/VulkanGraphicsMemoryAllocator.hpp>
+#include <NovelRT/Graphics/Vulkan/VulkanGraphicsResource.hpp>
+#include <NovelRT/Graphics/Vulkan/VulkanGraphicsResourceMemoryRegion.hpp>
 #include <NovelRT/Graphics/Vulkan/VulkanGraphicsTexture.hpp>
-#include <NovelRT/Exceptions/InitialisationFailureException.h>
+#include <NovelRT/Utilities/Macros.hpp>
+#include <NovelRT/Utilities/Span.hpp>
 
-namespace NovelRT::Graphics::Vulkan
+namespace NovelRT::Graphics
 {
-    VkImageView VulkanGraphicsTexture::CreateVulkanImageView()
+    using VulkanGraphicsDevice = GraphicsDevice<Vulkan::VulkanGraphicsBackend>;
+    using VulkanGraphicsDeviceObject = GraphicsDeviceObject<Vulkan::VulkanGraphicsBackend>;
+    using VulkanGraphicsMemoryAllocator = GraphicsMemoryAllocator<Vulkan::VulkanGraphicsBackend>;
+    using VulkanGraphicsResource = GraphicsResource<Vulkan::VulkanGraphicsBackend>;
+    template <template <typename> typename TResource>
+    using VulkanGraphicsResourceMemoryRegion = GraphicsResourceMemoryRegion<TResource, Vulkan::VulkanGraphicsBackend>;
+    using VulkanGraphicsTexture = GraphicsTexture<Vulkan::VulkanGraphicsBackend>;
+
+    VkImageView CreateVulkanImageView(
+        std::shared_ptr<GraphicsDevice<Vulkan::VulkanGraphicsBackend>> device,
+        GraphicsTextureKind kind,
+        VkImage image)
```

Comment:
This looks like quite a big API change. what happened?

Reply:
Mostly just hiding implementation details where they don't need to be public.

Reply:
Hmm, I think we want to be careful of this, but if you think its safe for now then honestly this PR is so big I can't really argue on it.